### ISSUE TITLE
Build issues on MacOS

### ISSUE THREAD

**Issue by dcoeurjo** [appleseedhq/appleseed#2064]
I have couple of issues trying to use/build appleseed on macOS (high sierra 10.13.4, up-to-date`brew` deps..). 

1. Latest appleseed mac version (1.9.0-beta) binaries give me some illegal instructions: 
```
./appleseed-studio
[1]    72959 illegal hardware instruction  ./appleseed-studio
```  
2. I tried to build the project following the macos instruction (I'll open a PR with some minor issues in the doc BTW) but there are issues with the python binding (full log: https://gist.github.com/dcoeurjo/5d21b1b4f7bd735ede11ce096867d5ba). I have a pretty standard install but I can share with you more details if you need.

(anyway, very cool project 👍 :))

**Comment by dcoeurjo** [comment]
:/ we cannot PR wiki pages apparently.

Quick diff: remove line 35 of https://github.com/appleseedhq/appleseed/wiki/Building-appleseed-on-macOS:   `brew tap homebrew/science`  (this tap has been closed and the `openimagio` formula has been moved to `homebrew/core`)

**Comment by dictoon** [comment]
Hey David,

> Latest appleseed mac version (1.9.0-beta) binaries give me some illegal instructions

I'm afraid this is the usual problem: I've compiled them with SSE4.2 support but your Mac doesn't support them. A _very_ common complain, unfortunately, and the way I'm building appleseed on macOS doesn't easily allow to completely disable SSE4.2 support...

> I tried to build the project following the macos instruction (I'll open a PR with some minor issues in the doc BTW) but there are issues with the python binding

Wow, I've never seen those errors. From the log you posted: 

    "boost::python::objects::make_nurse_and_patient(_object*, _object*)", referenced from:

??? :)

Anyway, are you sure you've been Boost.Python against Python 2.7 and that you're building appleseed against it as well? We don't support Python 3 yet.

> (anyway, very cool project 👍 :))

Thank you!!

**Comment by dictoon** [comment]
> Quick diff

Done, thanks!

**Comment by dcoeurjo** [comment]
mm... looks like my python is fine (2.7.15). Here you have the `cmake` output: https://gist.github.com/dcoeurjo/6d0194be5e4b9b5d54a8682f13f75c60

**Comment by dcoeurjo** [comment]
Ok I think I've spotted the issue...  :)

**Comment by dcoeurjo** [comment]
Fixed (cf #2071) ;) thx
